x-pack/plugins/serverless_observability/public/navigation_tree.ts

```diff
@@ -80,6 +81,7 @@ export const navigationTree: NavigationTreeDefinition = {
                 defaultMessage: 'Anomaly detection',
               }),
               link: 'ml:anomalyDetection',
+              id: 'ml:anomalyDetection',
```

Comment:
I added an ID to this link because now that we have more than one link in the same page, the test-subj values end up being the same as it uses the `link` value.  The wrong link then gets clicked.  I used the same value as the "deepLinkId" in `link` which I hope isn't confusing.